docs/manuals/userman/gui/pages/navigate/index.md

```diff
@@ -0,0 +1,285 @@
+        navigate(state, to=page)
+
+    pages = {
+        "/": root_md,
+        "home": home_md,
+        "about": about_md
+    }
+
+    Gui(pages=pages).run()
+    ```
+=== "Python"
+    ```python
+    from taipy import Gui, navigate
+    import taipy.gui.builder as tgb
+
+    def on_menu(state, action, info):
+        page = info["args"][0]
+        navigate(state, to=page)
+
+    with tgb.Page() as root_page:
+        tgb.menu(label="Menu",
+                lov=[('home', 'Page 1'), ('about', 'Page 2')],
+                on_action=on_menu)
+        tgb.text("# Multi-page application", mode="md")
+
+    with tgb.Page() as home_page:
+        tgb.text("# Home", mode="md")
+
+    with tgb.Page() as about_page:
+        tgb.text("# About", mode="md")
+
+    pages = {
+        "/": root_page,
+        "home": home_page,
+        "about": about_page
+    }
+
+    Gui(pages=pages).run()
+    ```
+
+# Using the `navigate` function
+
+The `navigate` function allows for programmatically controlling navigation 
+within callback functions. You can navigate to a page of this application or 
+an external page.
+
+=== "Markdown"
+    ```python
+    from taipy import Gui, navigate
+
+    root_md = """
+    <|Click to go to Page 1|button|on_action=go_home|>
+    # Multi-page application
+    """
+
+    home_md = "# Home"
+    about_md = "# About"
+
+    def go_home(state):
+        navigate(state, "home")
+
+    pages = {
+        "/": root_md,
+        "home": home_md,
+        "about": about_md
+    }
+
+    Gui(pages=pages).run()
+    ```
+=== "Python"
+    ```python
+    from taipy import Gui, navigate
+    import taipy.gui.builder as tgb
+
+    def go_home(state):
+        navigate(state, "home")
+
+    with tgb.Page() as root_page:
+        tgb.button(label="Click to go to Page 1", on_action=go_home)
+        tgb.text("# Multi-page application", mode="md")
+
+    with tgb.Page() as home_page:
+        tgb.text("# Home", mode="md")
+
+    with tgb.Page() as about_page:
+        tgb.text("# About", mode="md")
+
+    pages = {
+        "/": root_page,
+        "home": home_page,
+        "about": about_page
+    }
+
+    Gui(pages=pages).run()
+    ```
+
+
+# Using the `on_navigate` function
+
+The `on_navigate` function allows for custom control over navigation, such 
+as redirecting users based on conditions.
+
+See more information for [*on_navigate*](../../callbacks.md#navigation-callback).
+
+
+=== "Markdown"
+    ```python
+    from taipy import Gui, State
+
+    results_ready = False
+
+    def on_navigate(state: State, page_name: str):
+        if page_name == "results" and not state.results_ready:
+            return "no_results"
+        return page_name
+
+    root_md = "# Multi-page application"
+    results_md = "# Results Page"
+    no_results_md = "# No Results Available"
+
+    pages = {
+        "/": root_md,
+        "results": results_md,
+        "no_results": no_results_md
+    }
+
+    gui = Gui(pages=pages)
```

Comment:
```suggestion
    gui = Gui(pages=pages).run()
```

Reply:
Not sure about this one too. Do you want to erase the `gui = `then too?